extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,713 @@
+
+    this.barStick_.appendChild(this.hint_);
+  }
+
+  /**
+   * Install gestures for touch
+   * @private
+   */
+  registerTouchGestures_() {
+    if (this.gestures_) {
+      return;
+    }
+
+    this.gestures_ = Gestures.get(this.element);
+
+    this.gestures_.onGesture(SwipeXRecognizer, e => {
+      // We need the initial offset, yet gesture event seems not providing
+      if (e.data.first) {
+        // Disable hint reappearance timeout if needed
+        this.resetHintInterval_(true);
+      }
+      this.pointerMoveX_(
+          e.data.startX + e.data.deltaX);
+      if (e.data.last) {
+        // Reset hint reappearance timeout if needed
+        this.resetHintInterval_(!this.shouldHintLoop_);
+      }
+    });
+
+    this.gestures_.onPointerDown(e => {
+      // Ensure touchstart changes slider position
+      this.pointerMoveX_(e.touches[0].pageX);
+      // Use !this.shouldHintLoop here
+      // It is possible that after onPointerDown
+      // SwipeXRecognizer callback is not triggered
+      this.resetHintInterval_(!this.shouldHintLoop_);
+    });
+  }
+
+  /**
+   * Uninstall gestures for touch
+   * @private
+   */
+  unregisterTouchGestures_() {
+    if (!this.gestures_) {
+      return;
+    }
+    this.gestures_.cleanup();
+    this.gestures_ = null;
+  }
+
+  /**
+   * Reset interval when the hint would reappear
+   * Call this when an user interaction is done
+   * Specify opt_noRestart to true if no intend to start a timeout for
+   * showing hint again.
+   * @param {boolean=} opt_noRestart
+   * @private
+   */
+  resetHintInterval_(opt_noRestart) {
+    if (this.disableHint_) {
+      return;
+    }
+
+    if (!this.isHintHidden_) {
+      this.animateHideHint_();
+    }
+
+    if (this.hintTimeoutHandle_ !== null) {
+      Services.timerFor(this.win).cancel(this.hintTimeoutHandle_);
+    }
+
+    if (opt_noRestart === true) {
+      this.hintTimeoutHandle_ = null;
+      return;
+    }
+
+    // Use timer instead of default setTimeout
+    this.hintTimeoutHandle_ = Services.timerFor(this.win).delay(() => {
+      this.animateShowHint_();
+    }, this.hintInactiveInterval_);
+  }
+
+  /**
+   * Show hint with animation
+   * @private
+   */
+  animateShowHint_() {
+    const interpolate = numeric(
+        Number(getStyle(dev().assertElement(this.hint_), 'opacity')), 0.7);
+    this.isHintHidden_ = false;
+    return Animation.animate(dev().assertElement(this.hint_), v => {
+      setStyles(dev().assertElement(this.hint_), {
+        opacity: interpolate(v),
+      });
+    }, 200).thenAlways();
+  }
+
+  /**
+   * Hide hint with animation
+   * @private
+   */
+  animateHideHint_() {
+    const interpolate = numeric(
+        Number(getStyle(dev().assertElement(this.hint_), 'opacity')), 0);
+    return Animation.animate(dev().assertElement(this.hint_), v => {
+      setStyles(dev().assertElement(this.hint_), {
+        opacity: interpolate(v),
+      });
+    }, 200).then(() => this.isHintHidden_ = true);
```

Comment:
@cathyxz This action is deliberate, since I am complying with the rule that hint is not considered as hidden unless it is having `opacity: 0;`. Complying with this rule make logic a bit clearer.